torch/linalg/__init__.py

```diff
@@ -482,7 +482,8 @@
         - **solution** (*Tensor*): the least squares solution
         - **residuals** (*Tensor*):  if :math:`m > n` then for full rank matrices in :attr:`input` the tensor encodes
             the squared residuals of the solutions, that is :math:`||\text{input} @ x - b||_F^2`.
-            If :math:`m \le n`, an empty tensor is returned instead.
+            If :math:`m \le n` and rank is not equal to :math:`n`, an empty tensor is returned instead.
```

Comment:
The rank of what?

Reply:
I think it breaks the LAPACK interface. LAPACK returns residuals which only make sense under the full-rank assumption.  Shall we keep the LAPACK behavior or build on top of it?

If the full-rank assumption is violated, we could directly return `||ax - b||^2` instead of empty tensors/infinites...